diff -rup a/apps/req.c b/apps/req.c
--- a/apps/req.c	2019-09-10 09:13:07.000000000 -0400
+++ b/apps/req.c	2021-03-11 17:26:34.181716841 -0500
@@ -90,6 +90,7 @@ typedef enum OPTION_choice {
     OPT_VERIFY, OPT_NODES, OPT_NOOUT, OPT_VERBOSE, OPT_UTF8,
     OPT_NAMEOPT, OPT_REQOPT, OPT_SUBJ, OPT_SUBJECT, OPT_TEXT, OPT_X509,
     OPT_MULTIVALUE_RDN, OPT_DAYS, OPT_SET_SERIAL, OPT_ADDEXT, OPT_EXTENSIONS,
+    OPT_STARTDATE, OPT_ENDDATE, OPT_DATES,
     OPT_REQEXTS, OPT_PRECERT, OPT_MD,
     OPT_R_ENUM
 } OPTION_CHOICE;
@@ -131,6 +132,9 @@ const OPTIONS req_options[] = {
     {"subject", OPT_SUBJECT, '-', "Output the request's subject"},
     {"multivalue-rdn", OPT_MULTIVALUE_RDN, '-',
      "Enable support for multivalued RDNs"},
+    {"startdate", OPT_STARTDATE, 's', "Cert notBefore, YYMMDDHHMMSSZ"},
+    {"enddate", OPT_ENDDATE, 's',
+     "YYMMDDHHMMSSZ cert notAfter (overrides -days)"},
     {"days", OPT_DAYS, 'p', "Number of days cert is valid for"},
     {"set_serial", OPT_SET_SERIAL, 's', "Serial number to use"},
     {"addext", OPT_ADDEXT, 's',
@@ -230,6 +234,7 @@ int req_main(int argc, char **argv)
     char *nofree_passin = NULL, *nofree_passout = NULL;
     char *req_exts = NULL, *subj = NULL;
     char *template = default_config_file, *keyout = NULL;
+    char *startdate = NULL, *enddate = NULL;
     const char *keyalg = NULL;
     OPTION_CHOICE o;
     int ret = 1, x509 = 0, days = 0, i = 0, newreq = 0, verbose = 0;
@@ -365,6 +370,12 @@ int req_main(int argc, char **argv)
         case OPT_X509:
             x509 = 1;
             break;
+        case OPT_STARTDATE:
+            startdate = opt_arg();
+            break;
+        case OPT_ENDDATE:
+            enddate = opt_arg();
+            break;
         case OPT_DAYS:
             days = atoi(opt_arg());
             break;
@@ -422,6 +433,10 @@ int req_main(int argc, char **argv)
 
     if (days && !x509)
         BIO_printf(bio_err, "Ignoring -days; not generating a certificate\n");
+    if (startdate != NULL && !x509)
+        BIO_printf(bio_err, "Ignoring -startdate; not generating a certificate\n");
+    if (enddate != NULL && !x509)
+        BIO_printf(bio_err, "Ignoring -enddate; not generating a certificate\n");
     if (x509 && infile == NULL)
         newreq = 1;
 
@@ -734,12 +749,36 @@ int req_main(int argc, char **argv)
 
             if (!X509_set_issuer_name(x509ss, X509_REQ_get_subject_name(req)))
                 goto end;
+
+            if (startdate != NULL && !ASN1_TIME_set_string_X509(NULL, startdate)) {
+                BIO_printf(bio_err,
+                           "start date is invalid, it should be YYMMDDHHMMSSZ or YYYYMMDDHHMMSSZ\n");
+                goto end;
+            }
+
+            if (enddate != NULL && !ASN1_TIME_set_string_X509(NULL, enddate)) {
+                BIO_printf(bio_err,
+                           "end date is invalid, it should be YYMMDDHHMMSSZ or YYYYMMDDHHMMSSZ\n");
+                goto end;
+            }
+
             if (days == 0) {
                 /* set default days if it's not specified */
                 days = 30;
             }
-            if (!set_cert_times(x509ss, NULL, NULL, days))
+
+            if (!set_cert_times(x509ss, startdate, enddate, days))
                 goto end;
+
+            if (enddate != NULL) {
+                int tdays;
+
+                if (!ASN1_TIME_diff(&tdays, NULL, NULL,
+                    X509_get0_notAfter(x509ss)))
+                    goto end;
+                days = tdays;
+            }
+
             if (!X509_set_subject_name
                 (x509ss, X509_REQ_get_subject_name(req)))
                 goto end;
diff -rup a/doc/man1/req.pod b/doc/man1/req.pod
--- a/doc/man1/req.pod	2019-09-10 09:13:07.000000000 -0400
+++ b/doc/man1/req.pod	2021-03-11 17:26:34.181716841 -0500
@@ -34,6 +34,8 @@ B<openssl> B<req>
 [B<-config filename>]
 [B<-multivalue-rdn>]
 [B<-x509>]
+[B<-startdate date>]
+[B<-enddate date>]
 [B<-days n>]
 [B<-set_serial n>]
 [B<-newhdr>]
@@ -247,6 +249,22 @@ the serial number.
 If existing request is specified with the B<-in> option, it is converted
 to the self signed certificate otherwise new request is created.
 
+=item B<-startdate date>
+
+When the B<-x509> option is being used this allows the start date to
+be explicitly set. The format of the date is YYMMDDHHMMSSZ (the same
+as an ASN1 UTCTime structure), or YYYYMMDDHHMMSSZ (the same as an ASN1
+GeneralizedTime structure). In both formats, seconds SS and timezone Z
+must be present.
+
+=item B<-enddate date>
+
+When the B<-x509> option is being used this allows the expiry date to
+be explicitly set. The format of the date is YYMMDDHHMMSSZ (the same
+as an ASN1 UTCTime structure), or YYYYMMDDHHMMSSZ (the same as an ASN1
+GeneralizedTime structure). In both formats, seconds SS and timezone Z
+must be present.
+
 =item B<-days n>
 
 When the B<-x509> option is being used this specifies the number of
